extensions/omni_txn/omni_txn.c

```diff
@@ -51,17 +75,20 @@ Datum retry(PG_FUNCTION_ARGS) {
       ErrorData *err = CopyErrorData();
       if (err->sqlerrcode == ERRCODE_T_R_SERIALIZATION_FAILURE) {
         ereport(NOTICE, errmsg("%d", max_attempts));
-        if (++retry_attempts > max_attempts) {
-          ereport(ERROR, errcode(err->sqlerrcode),
-                  errmsg("maximum number of retries (%d) has been attempted", max_attempts));
-        }
         PopActiveSnapshot();
         AbortCurrentTransaction();
 
         SetCurrentStatementStartTimestamp();
         StartTransactionCommand();
         PushActiveSnapshot(GetTransactionSnapshot());
 
+        if (++retry_attempts > max_attempts) {
+          ereport(ERROR, errcode(err->sqlerrcode),
```

Comment:
Any reason why this block moved down? I kept it up so that we don't restart a transaction too early.

Reply:
I didn't want to sleep while holding resources. But you are right, the throwing of errors should be done if the max_attempts is crossed. Will fix